synfig-studio/src/gui/instance.cpp

```diff
@@ -274,30 +283,54 @@ studio::Instance::run_plugin(std::string plugin_path)
 			one_moment.show();
 
 		} else {
-			bool result;
-			result = launcher.execute( plugin_path, App::get_base_path() );
+			
+			
+			// Save file copy
+			FileSystem::ReadStream::Handle stream_in = temporary_filesystem->get_read_stream("#project.sifz");
+			if (!stream_in)
+			{
+				synfig::error("run_plugin(): Unable to open file for reading");
+			}
+			stream_in = new ZReadStream(stream_in);
+			std::ofstream  outfile(filename_processed, std::ios::binary);
+			outfile << stream_in->rdbuf();
+			outfile.close();
+			
+			bool result=true;
+			//result = launcher.execute( plugin_path, App::get_base_path() );
 			if (!result){
 				one_moment.hide();
 				App::dialog_message_1b(
 						"Error",
-						launcher.get_output(),
+						"", //launcher.get_output(),
 						"details",
 						_("Close"));
 
 				one_moment.show();
 
+			} else {
+			
+				// Restore file copy
+				FileSystem::WriteStream::Handle stream = temporary_filesystem->get_write_stream(tmp_filename);
+				if (!stream)
+				{
+					synfig::error("run_plugin(): Unable to open file for write");
+				}
+				stream = FileSystem::WriteStream::Handle(new ZWriteStream(stream));
+				std::ifstream  infile(filename_processed, std::ios::binary);
+				*stream << infile.rdbuf();
```

Comment:
@blackwarthog Can you please check - if this is correct way to copy-and-compress file?